### PR TITLE
[ENH] Consolidate stats unit tests and fix linting errors in tests

### PR DESCRIPTION
**Pull request by jbteves**
Closes None

Changes proposed in this pull request:

- consolidates unit tests for modules into single files
- adds linting to /tests

### CODE REVIEW

File: tedana/tests/test_stats.py [hunk 1]
```diff
@@ -1,13 +1,80 @@
...
 import numpy as np
 import pytest
+import random
 
+from tedana.stats import computefeats2
 from tedana.stats import get_coeffs
+from tedana.stats import getfbounds
+
+
+"""
```
Comment (emdupre) [inline tedana/tests/test_stats.py]: I'm a little confused having these block comments outside of the functions. Is there a strong reason to have them, here ?

File: tedana/tests/test_stats.py [hunk 2]
```diff
@@ -1,13 +1,80 @@
...
+    Ensure that computefeats2 fails when input data do not have the right
+    shapes.
+    """
+    n_samples, n_vols, n_comps = 10000, 100, 50
+    data = np.empty((n_samples, n_vols))
+    mmix = np.empty((n_vols, n_comps))
+    mask = np.empty((n_samples))
 
+    data = np.empty((n_samples))
+    with pytest.raises(ValueError) as e_info:
```
Comment (emdupre) [inline tedana/tests/test_stats.py]: why not just have this as a straight assertion ?

File: tedana/tests/test_stats.py [hunk 3]
```diff
@@ -1,13 +1,80 @@
...
+    mask = np.empty((n_samples))
+    mmix = np.empty((n_vols+1, n_comps))
+    with pytest.raises(ValueError) as e_info:
+        computefeats2(data, mmix, mask, normalize=True)
+    assert str(e_info.value) == ('Second dimensions (number of volumes) of data ({0}) '
+                                 'and mmix ({1}) do not match.'.format(data.shape[0],
+                                                                       mmix.shape[0]))
 
+
+# SMOKE TEST
```
Comment (emdupre) [inline tedana/tests/test_stats.py]:
```suggestion

```

File: tedana/tests/test_stats.py [hunk 4]
```diff
@@ -94,3 +161,27 @@ def test_smoke_get_coeffs():
...
+Tests for tedana.stats.getfbounds
+"""
+def test_getfbounds():
+    good_inputs = range(1, 12)
+
+    for n_echos in good_inputs:
+        getfbounds(n_echos)
+
+
+# SMOKE TEST
```
Comment (emdupre) [inline tedana/tests/test_stats.py]:
```suggestion

```

File: tedana/tests/test_stats.py (hunk 2 above)
Comment (emdupre) [reply tedana/tests/test_stats.py]: Actually, `pytest.raise` would make more sense: http://doc.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

**Comment by jbteves** [comment]
Hmm, I thought the linter was supposed to run on unit tests, that should have been caught. Let me investigate.

**Pull request merged**